### PR TITLE
ML Translation

### PR DESCRIPTION
**Pull request by tiananguyen**
This feature translates messages into another language. 
Right now I'm getting the package com.google.cloud.translate does not exist.

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 1]
```diff
@@ -92,11 +102,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+  /**
    * Private method to recognize the image url in the text message and convert to <img> tag.
    * Message with no img url will not be modified.
    */
   private String extractImgUrl(String text) {
     return (text.replaceAll(this.imgUrlRegex, this.imgUrlReplacement));
   }
+
+  /**
+   * Translates messages based on a URL query parameter.
```
Comment (maksymko) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: This comment is misleading. You are not doing it based on the URL parameter, you are doing it based on the argument value.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 2]
```diff
@@ -92,11 +102,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+   */
+  private void translateMessages(List<Message> messages, String targetLanguageCode) {
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
```
Comment (maksymko) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: Does it handle the case where the source and target languages are the same?

File: src/main/webapp/js/user-page-loader.js [hunk 3]
```diff
@@ -91,6 +98,22 @@ function buildMessageDiv(message) {
   return messageDiv;
 }
 
+/** Provides link to URL */
+function buildLanguageLinks(){
+  const userPageUrl = '/user-page.html?user=' + parameterUsername;
+  const languagesListElement  = document.getElementById('languages');
+  languagesListElement.appendChild(createListItem(createLink(
```
Comment (maksymko) [inline src/main/webapp/js/user-page-loader.js]: Did you forget to define `createListItem` and `createLink`? Or what is happening here?

In any case, a lot of repetition.
define a list or a map of languages, something like
```
const languages = [['English', 'en'], ['Chinese', 'zh'], ['Hindi', 'hi']...]
```
and then iterate over it in the loop, creating the needed links.

**Review by maksymko — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/user-page-loader.js (hunk 3 above)
Comment (tiananguyen) [reply src/main/webapp/js/user-page-loader.js]: Sounds good! 
The functions were from a different file, navigation-loader.js.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 2 above)
Comment (tiananguyen) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: Yes! It should. I'll test that.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 1 above)
Comment (tiananguyen) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: That's right, I'll fix that thank you!

**Review by maksymko — APPROVED** [review]

**Pull request merged**